parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -293,21 +287,44 @@ setupCNIDirs() {
     chmod 755 $CNI_CONFIG_DIR
 }
 
-# For CNI/AzureCNI, we want to use the untar azurecni reference first. And if that doesn't exist on the vhd does the tgz?
-# And if tgz is already on the vhd then just untar into CNI_BIN_DIR
-# Latest VHD should have the untar, older should have the tgz. And who knows will have neither.
+
+# Reference CNI plugins is used by kubenet and the loopback plugin used by containerd 1.0 (dependency gone in 2.0)
+# The version used to be deteremined by RP/toggle but are now just hadcoded in vhd as they rarely change and require a node image upgrade anyways
+# Latest VHD should have the untar, older should have the tgz. And who knows will have neither. 
 installCNI() {
-    CNI_TGZ_TMP=${CNI_PLUGINS_URL##*/} # Use bash builtin ## to remove all chars ("*") up to the final "/"
-    CNI_DIR_TMP=${CNI_TGZ_TMP%.tgz}    # Use bash builtin % to remove the .tgz to look for a folder rather than tgz
+    #always just use what is listed in components.json so we don't have to sync.
+    cniPackage=$(jq ".Packages" "$COMPONENTS_FILEPATH" | jq ".[] | select(.name == \"cni-plugins\")") || exit $ERR_CNI_VERSION_INVALID
+    
+    #CNI doesn't really care about this but wanted to reuse returnPackageVersions which requires it.
+    os=${UBUNTU_OS_NAME} 
+    if [[ -z "$UBUNTU_RELEASE" ]]; then
+        os=${MARINER_OS_NAME}
+        os_version="current"
+    fi
+    os_version="${UBUNTU_RELEASE}"
+    PACKAGE_VERSIONS=()
+    returnPackageVersions "${cniPackage}" "${os}" "${os_version}"
+    
+    #should change to ne
+    if [[ ${#PACKAGE_VERSIONS[@]} -gt 1 ]]; then
+        echo "WARNING: containerd package versions array has more than one element. Installing the last element in the array."
+        exit $ERR_CONTAINERD_VERSION_INVALID
+    fi
+    packageVersion=${PACKAGE_VERSIONS[0]}
 
+    # Is there a ${arch} variable I can use instead of the iff
```

Comment:
from `installCrictl`: `CPU_ARCH=$(getCPUArch)`, feel free to make that a global var :)

Reply:
seems `CPU_ARCH` will already be defined at the global scope during build, though I guess `installCNI` will only ever be called at runtime after Devin's changes